### DISCUSSION TITLE
Use of SourceLink in the nuget package

### DISCUSSION THREAD

**Discussion by YannickRe** [pnp/pnpframework#81] [Ideas] [answered]
I just learned about the existence of SourceLink, which is essentially a technology that allows us to step through third party source code when debugging. It would be extremely useful for developers to step through the source code of PnP.Framework and PnP.Core, while having issues occurring.

It would increase developer experiences, helping us to root cause issues with our own code leading to exceptions in PnP.Framework for example. Additionally; this might reduce number of questions AND lead to higher engagement on fixing bugs. Devs would more easily see the error occurring while debugging their code, which might trigger a PR instead of a bug ticket.

More info on SourceLink: https://github.com/dotnet/sourcelink

What do you think?

**Comment by jansenbe** [comment]
I have been thinking about it in the past, makes sense to do this.

**Comment by jansenbe** [comment, marked as answer]
@YannickRe : here you go: https://www.nuget.org/packages/PnP.Framework/0.1.113-preview

**Comment by YannickRe** [comment]
You are awesome! This will help us debug and potentially fix issues a lot quicker.
Two more questions:
- Will you be including some docs somewhere on how to turn this on in Visual Studio? Might make sense to "promote" this to developers.
- Will you do this for PnP.Core too? Or should I go ask it over there? ;)

**Reply by jansenbe** [reply]
https://pnp.github.io/pnpcore/using-the-sdk/basics-debug.html

**Reply by YannickRe** [reply]
You are the best! Also, you should set up sponsorship on GitHub so I can sponsor you for this awesome work ;)
